### PR TITLE
doc/datasource/vmware: Update contents relevant to disable_vmware_customization configuration

### PR DESCRIPTION
**Pull request by PengpengSun**
## Proposed Commit Message

```
summary: Update contents relevant to disable_vmware_customization configuration

1. Let customer know that VMware Tools only checks 'disable_vmware_customization'
   configuration in cloud-init configuration file /etc/cloud/cloud.cfg by far.
2. we call it 'Guest OS Customization' but 'Guest OS Customisation', correct the word.
```

## Checklist:

 - [x] My code follows the process laid out in [the documentation](https://cloudinit.readthedocs.io/en/latest/development/contributing.html)
 - [x] I have updated or added any unit tests accordingly
 - [x] I have updated or added any documentation accordingly

### CODE REVIEW

**Review by blackboxsw — COMMENTED** [review]
Thank you @PengpengSun for the PR. 

It is nice to standardize on terminology that matches the configuration key name ``disable_vmware_customization``. Generally with official docs in Canonical projects typically follow British English spellings instead of US English per these guidelines:
  * [style guidance for cloud-init](https://cloudinit.readthedocs.io/en/latest/development/docs.html#style-guide)
  * [style guidance for Ubuntu projects](https://docs.ubuntu.com/styleguide/en/)

But, I think this is an unique case, for the VMware datasource that may warrant using "customization" throughout because VMware's own external docs, which are linked to from this page make abundant use of US English `customization` operations in VMware products.

I've set the `documentation` label on this PR to get doc-team input here.

File: doc/rtd/reference/datasources/vmware.rst [hunk 1]
```diff
@@ -441,8 +454,8 @@ the metadata key ``network``. Valid encodings are ``base64`` and
 .. LINKS
 .. _Guest OS Customization: https://docs.vmware.com/en/VMware-vSphere/8.0/vsphere-vm-administration/GUID-58E346FF-83AE-42B8-BE58-253641D257BC.html
 .. _GuestInfo keys: https://github.com/vmware/govmomi/blob/master/govc/USAGE.md
-.. _traditional Linux customisation specification: https://docs.vmware.com/en/VMware-vSphere/8.0/vsphere-vm-administration/GUID-EB5F090E-723C-4470-B640-50B35D1EC016.html#GUID-9A5093A5-C54F-4502-941B-3F9C0F573A39__GUID-40C60643-A2EB-4B05-8927-B51AF7A6CC5E
-.. _Guest customisation using cloud-init: https://developer.vmware.com/docs/17020/vsphere-web-services-sdk-programming-guide--8-0-/GUID-75E27FA9-2E40-4CBF-BF3D-22DCFC8F11F7.html
+.. _traditional Linux customization specification: https://docs.vmware.com/en/VMware-vSphere/8.0/vsphere-vm-administration/GUID-EB5F090E-723C-4470-B640-50B35D1EC016.html#GUID-9A5093A5-C54F-4502-941B-3F9C0F573A39__GUID-40C60643-A2EB-4B05-8927-B51AF7A6CC5E
+.. _Guest customization using cloud-init: https://developer.vmware.com/docs/17020/vsphere-web-services-sdk-programming-guide--8-0-/GUID-75E27FA9-2E40-4CBF-BF3D-22DCFC8F11F7.html
```
Comment (blackboxsw) [inline doc/rtd/reference/datasources/vmware.rst:458]: This link is not publicly visible to me when I click it. Do we have the right link?

**Review by blackboxsw — CHANGES_REQUESTED** [review]
@PengpengSun I know you didn't change the actual link for "Guest customization using cloud-init" in this PR. But, in trying to click through the links to test. I get 

`The document does not exist or you have no permission to view it.`
from https://developer.vmware.com/docs/17020/vsphere-web-services-sdk-programming-guide--8-0-/GUID-75E27FA9-2E40-4CBF-BF3D-22DCFC8F11F7.html


Is there a different link that we can provide that is publicly accessible?

File: doc/rtd/reference/datasources/vmware.rst (hunk 1 above)
Comment (PengpengSun) [reply doc/rtd/reference/datasources/vmware.rst]: Hi @blackboxsw I've updated the link in a new commit, please take a look, thanks.

**Review by blackboxsw — APPROVED** [review]
Thanks for the correction @PengpengSun

**Pull request merged**